libraries/lib-wave-track/WaveTrack.h

```diff
@@ -1369,6 +1371,17 @@ class WAVE_TRACK_API WaveTrackFactory final
    SampleBlockFactoryPtr mpFactory;
 };
 
+//! TrackAttachments that implement this are notified of changes in the list
+//! of clips
+class WAVE_TRACK_API ClipListListener
```

Comment:
How would I create the observers and the subscriptions if WaveTrack is the publisher?

Each new WaveTrack must inform its own listener, which has its own separate state (the wave track color), of the event of addition of clips.

So the listener must be created too with each WaveTrack -- and that's where `ClientData` is useful, if also I use `BuildAll` instead of creation on demand.  It allows hooking of a constructor of a class.

And I need `BuildAll` so that the new attached object is present from the beginning, before any clips are created or deserialized.

But you are right that I could then also use Observer instead of this class.

There is the problem of getting deep-copy right, but that is what `Reparent` is for.